utils/terrascript_client.py

```diff
@@ -344,6 +352,58 @@ def populate_additional_providers(self, accounts):
                            alias=alias,
                            assume_role={'role_arn': assume_role})
 
+    def populate_route53(self, desired_state, default_ttl=300):
+        for zone in desired_state:
+            acct_name = zone['account_name']
+
+            # Ensure zone is in the state for the given account
+            zone_id = safe_resource_id(f"{zone['name']}")
+            zone_values = {
+                'name': zone['name'],
+                'comment': 'Managed by Terraform'
+            }
+            zone_resource = aws_route53_zone(zone_id, **zone_values)
+            self.add_resource(acct_name, zone_resource)
+
+            counts = {}
+            for record in zone['records']:
+                record_fqdn = f"{record['name']}.{zone['name']}"
+                record_id = safe_resource_id(
+                    f"{record_fqdn}_{record['type'].upper()}")
+
+                # Count record names so we can generate unique IDs
+                if record_id not in counts:
+                    counts[record_id] = 0
+                counts[record_id] += 1
+
+                # If more than one record with a given name, append _{count}
+                if counts[record_id] > 1:
+                    record_id = f"{record_id}_{counts[record_id]}"
```

Comment:
Is it important that this can change if a new record with a given name is added after we already have one? there's little we can do to avoid it...

Reply:
Only the 2nd, 3rd, etc... records are going to have a number appended. The original record won't have its name changes and thus shouldn't cause a resource removal/addition.

Did I accurately understand your point or did you mean something different?

Reply:
Yes, thanks